test/common/run-tests

```diff
@@ -148,6 +161,11 @@ def run(opts, image):
     # Make sure tests can make relative imports
     sys.path.append(os.path.realpath(opts.test_dir))
 
+    # Build list of affected tests
+    cmd = ["git", "diff", "--name-only", "HEAD", "origin/master", opts.test_dir]
```

Comment:
admittedly I haven't tested this with `make-checkout` and the full CI machinery. Running tests here, if it does not work, then I'll check how CI prepared checkout looks like.

Reply:
We must also make sure that this works with e. g. a bots PR, where that diff should be empty. Or a downstream gating test where the `git diff` may even fail as the directory isn't even a git  checkout, but an unpacked release tarball.

Reply:
Fixed this so we look at return code of the `git` command. (The same way we do it with different git command in the same file).